src/hooks/useTokenBalances.ts

```diff
@@ -57,13 +60,20 @@ async function _getBalances(): Promise<TokenBalanceDetails[]> {
 }
 
 export const useTokenBalances = (): UseTokenBalanceResult => {
+  const walletInfo = useWalletConnection()
   const [balances, setBalances] = useState<TokenBalanceDetails[] | undefined>(null)
   const [error, setError] = useState(false)
   const mounted = useRef(true)
 
   useEffect(() => {
-    _getBalances()
-      .then(balances => setBalances(balances))
+    _getBalances(walletInfo)
+      .then(balances => {
+        console.log(
+          '[useTokenBalances] Wallet balances',
+          balances ? balances.map(b => formatAmount(b.walletBalance)) : null,
+        )
+        setBalances(balances)
```

Comment:
I thought you were going to change mounted check here, or was that in a different PR?

Reply:
different PR